src/device/gcn/output.jl

```diff
@@ -156,6 +156,26 @@ function wave_serialized(func::Function)
     return result
 end
 
+const SC_PAGESIZE = 30
+const MS_ASYNC = 1
+@static if Sys.islinux()
+const PAGESIZE = ccall(:sysconf, Clong, (Cint,), SC_PAGESIZE)
+end
+function semi_safe_load(ptr::Ptr{T}) where T
+    num_pages = ceil(Int, sizeof(T) / PAGESIZE)
```

Comment:
What if the object spans two pages, but it itself is `< page`?

Reply:
Latest commit should handle this